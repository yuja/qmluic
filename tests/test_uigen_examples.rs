pub mod common;

#[test]
fn test_gadget_properties() {
    insta::assert_snapshot!(common::translate_file("examples/GadgetProperties.qml").unwrap());
}

#[test]
fn test_item_views() {
    insta::assert_snapshot!(common::translate_file("examples/ItemViews.qml").unwrap());
}

#[test]
fn test_layout_flow() {
    insta::assert_snapshot!(common::translate_file("examples/LayoutFlow.qml").unwrap());
}

#[test]
fn test_main_window() {
    insta::assert_snapshot!(common::translate_file("examples/MainWindow.qml").unwrap());
}

#[test]
fn test_settings_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/SettingsDialog.qml").unwrap());
}

#[test]
fn test_static_item_model() {
    insta::assert_snapshot!(common::translate_file("examples/StaticItemModel.qml").unwrap());
}

#[test]
fn test_various_layouts() {
    insta::assert_snapshot!(common::translate_file("examples/VariousLayouts.qml").unwrap());
}

#[test]
fn test_customwidget_common_my_dialog_button_box() {
    insta::assert_snapshot!(common::translate_file(
        "examples/customwidget/common/MyDialogButtonBox.qml"
    )
    .unwrap());
}

#[test]
fn test_customwidget_main_dialog() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/MainDialog.qml").unwrap()
    );
}

#[test]
fn test_customwidget_settings_form() {
    insta::assert_snapshot!(
        common::translate_file("examples/customwidget/SettingsForm.qml").unwrap()
    );
}

#[test]
fn test_thg_hg_email_dialog() {
    insta::assert_snapshot!(common::translate_file("examples/thg/HgEmailDialog.qml").unwrap());
}
