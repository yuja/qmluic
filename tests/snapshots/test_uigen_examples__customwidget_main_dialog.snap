---
source: tests/test_uigen_examples.rs
expression: "common::translate_file(\"examples/customwidget/MainDialog.qml\").unwrap()"
---
<ui version="4.0">
 <class>MainDialog</class>
 <widget class="QDialog">
  <layout class="QVBoxLayout">
   <item>
    <widget class="SettingsForm" name="settingsForm">
    </widget>
   </item>
   <item>
    <widget class="MyDialogButtonBox" name="buttonBox">
    </widget>
   </item>
   <item>
    <widget class="MyDialogButtonBox" name="buttonBox2">
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SettingsForm</class>
   <extends>QWidget</extends>
   <header>settingsform.h</header>
  </customwidget>
  <customwidget>
   <class>MyDialogButtonBox</class>
   <extends>QDialogButtonBox</extends>
   <header>mydialogbuttonbox.h</header>
  </customwidget>
 </customwidgets>
</ui>

